<launch>
    <arg name="video_device"/>
    <arg name="map_path" default="$(find comp4)/data/map.yaml"/>

    <include file="$(find turtlebot_bringup)/launch/minimal.launch"/>
    <include file="$(find turtlebot_teleop)/launch/logitech.launch"/>
    <include file="$(find turtlebot_rviz_launchers)/launch/view_navigation.launch"/>

    <include file="$(find comp4)/launch/amcl.launch">
        <arg name="map_file" value="$(arg map_path)"/>
        <param file="$(find comp4)/param/costmap_common_params.yaml" command="load" ns="global_costmap" />
        <param file="$(find comp4)/param/costmap_common_params.yaml" command="load" ns="local_costmap" />
        <param file="$(find comp4)/param/global_costmap_params.yaml" command="load" />
        <param file="$(find comp4)/param/local_costmap_params.yaml" command="load" />
        <param file="$(find comp4)/param/dwa_local_planner_params.yaml" command="load" />
        <param file="$(find comp4)/param/move_base_params.yaml" command="load" />
    </include>

    <include file="$(find comp4)/launch/ar_track.launch"/>

    <node name="red_line_finder" pkg="comp4" type="red_line_finder.py"/>
    <node name="white_line_finder" pkg="comp4" type="white_line_tracker.py" />
    <node name="white_line_ramp" pkg="comp4" type="white_line_ramp.py" />

    <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
        <param name="video_device" value="$(arg video_device)" />
        <param name="image_width" value="640" />
        <param name="image_height" value="480" />
        <param name="pixel_format" value="yuyv" />
        <param name="camera_frame_id" value="usb_cam" />
        <param name="io_method" value="mmap"/>
        <param name="autofocus" value="false"/>
        <param name="autoexposure" value="false"/>
        <param name="auto_white_balance" value="false"/>
        <param name="contrast" value="100"/>
    </node>

    <param name="/move_base/global_costmap/inflation_layer/inflation_radius" value="0.07"/>
    <param name="/move_base/local_costmap/inflation_layer/inflation_radius" value="0.07"/>
    
    <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
        <remap from="image" to="/usb_cam/image_raw"/>
        <param name="autosize" value="true" />
    </node>
</launch>
